core/src/main/java/com/imaginea/kodebeagle/GitHelper/ListFileContents.java

```diff
@@ -0,0 +1,432 @@
+package com.imaginea.kodebeagle.GitHelper;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.diff.DiffEntry;
+import org.eclipse.jgit.diff.DiffEntry.ChangeType;
+import org.eclipse.jgit.diff.RawTextComparator;
+import org.eclipse.jgit.errors.LargeObjectException;
+import org.eclipse.jgit.lib.Constants;
+import org.eclipse.jgit.lib.FileMode;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.ObjectLoader;
+import org.eclipse.jgit.lib.PersonIdent;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevObject;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.treewalk.TreeWalk;
+
+import com.imaginea.kodebeagle.GitHelper.PathModel.PathChangeModel;
+
+public class ListFileContents {
+
+	/**
+	 */
+	public static Map<String, Integer> getNRelatedFiles(Repository repository, String filePath, int n)
+			throws GitAPIException, IOException {
+
+		filePath = "core/src/main/scala/com/kodebeagle/crawler/ZipBasicParser.scala";
+
+		Map<String, Integer> fileRankings = new HashMap<String, Integer>();
+
+		walkAllCommits(repository, fileRankings, filePath);
+
+		Map<String, Integer> topNFiles = SortRankings.sortRankingsMap(fileRankings, n);
+		return topNFiles;
+	}
+
+	/**
+	 * This method uses RevWalk to quickly iterate over all available commits
+	 * 
+	 * @param repo
+	 * @throws GitAPIException
+	 * @throws IOException
+	 */
+	public static void walkAllCommits(Repository repo, Map<String, Integer> fileRankings, String fileName)
+			throws GitAPIException, IOException {
+
+		Git git = new Git(repo);
+		RevWalk walk = new RevWalk(repo);
+		List<Ref> branches = git.branchList().call();
+
+		for (Ref branch : branches) {
+			String branchName = branch.getName();
+
+			Iterable<RevCommit> commits = git.log().all().call();
+
+			for (RevCommit commit : commits) {
+				boolean foundInThisBranch = false;
+
+				RevCommit targetCommit = walk.parseCommit(repo.resolve(commit.getName()));
+				for (Entry<String, Ref> e : repo.getAllRefs().entrySet()) {
+					if (e.getKey().startsWith("refs/heads/")) {
+						if (walk.isMergedInto(targetCommit, walk.parseCommit(e.getValue().getObjectId()))) {
+							String foundInBranch = e.getValue().getName();
+							if (branchName.equals(foundInBranch)) {
+								foundInThisBranch = true;
+								break;
+							}
+						}
+					}
+				}
+				if (foundInThisBranch) {
```

Comment:
Why iterating over all the branched and yet doing the same thing for all commits regardless of their branch?